File: specs/typespec/org/hub-profiles.ts
import type { ValidationError } from "../common/common";
import { newValidationError } from "../common/common";
import type { Handle } from "../hub/hub-users";
import type { HubProfilePublicView } from "../hub/profile";
import type { PublicEmployerStint } from "../hub/work-emails";

export interface OrgGetHubUserProfileRequest {
	handle: Handle;
}

export interface OrgHubUserProfileResponse {
	profile: HubProfilePublicView;
	stints: PublicEmployerStint[];
}

export function validateOrgGetHubUserProfileRequest(
	req: unknown
): ValidationError[] {
	const errors: ValidationError[] = [];
	if (!req || typeof req !== "object") {
		return [{ field: "$root", message: "Request body is required" }];
	}
	const r = req as Record<string, unknown>;
	if (!r.handle || typeof r.handle !== "string" || r.handle.trim() === "") {
		errors.push(newValidationError("handle", "Handle is required"));
	}
	return errors;
}

File: api-schema/org/hub-profiles.go
package org

import (
	"vetchium-api-server.typespec/common"
	hub "vetchium-api-server.typespec/hub"
)

type OrgGetHubUserProfileRequest struct {
	Handle string `json:"handle"`
}

type OrgHubUserProfileResponse struct {
	Profile hub.HubProfilePublicView  `json:"profile"`
	Stints  []hub.PublicEmployerStint `json:"stints"`
}

func (r OrgGetHubUserProfileRequest) Validate() []common.ValidationError {
	var errs []common.ValidationError
	if r.Handle == "" {
		errs = append(errs, common.ValidationError{
			Field:   "handle",
			Message: "Handle is required",
		})
	}
	return errs
}
